app/src/screens/document/IDPickerScreen.tsx

```diff
@@ -0,0 +1,211 @@
+import PlusIcon from '@/images/icons/plus.svg';
+import SelfLogo from '@/images/logo.svg';
+import { useSafeAreaInsets } from '@/mocks/react-native-safe-area-context';
+import { black, slate100, slate300, slate400, white } from '@/utils/colors';
+import { extraYPadding } from '@/utils/constants';
+import { advercase, dinot } from '@/utils/fonts';
+import { buttonTap } from '@/utils/haptic';
+
+type DocumentStackParamList = {
+  IDPicker: {
+    countryCode: string;
+    documentTypes: string[];
+    countryName: string;
+  };
+};
+
+type IDPickerScreenRouteProp = RouteProp<DocumentStackParamList, 'IDPicker'>;
+
+const getDocumentName = (docType: string): string => {
+  switch (docType) {
+    case 'p':
+      return 'Passport';
+    case 'i':
+      return 'ID card';
+    case 'a':
+      return 'Aadhaar';
+    default:
+      return 'Unknown Document';
+  }
+};
+
+const getDocumentNameForEvent = (docType: string): string => {
+  switch (docType) {
+    case 'p':
+      return 'passport';
+    case 'i':
+      return 'id_card';
+    case 'a':
+      return 'aadhaar';
+    default:
+      return 'unknown_document';
+  }
+};
+
+const getDocumentDescription = (docType: string): string => {
+  switch (docType) {
+    case 'p':
+      return 'Verified Biometric Passport';
+    case 'i':
+      return 'Verified Biometric ID card';
+    case 'a':
+      return 'Verified mAadhaar QR code';
+    default:
+      return 'Unknown Document';
+  }
+};
+
+const getDocumentLogo = (docType: string): React.ReactNode => {
+  switch (docType) {
+    case 'p':
+      return <EPassportLogoRounded />;
+    case 'i':
+      return <EPassportLogoRounded />;
+    case 'a':
+      return <AadhaarLogo />;
+    default:
+      return null;
+  }
+};
+
+const IDPickerScreen: React.FC = () => {
+  const route = useRoute<IDPickerScreenRouteProp>();
+  const { countryCode, documentTypes } = route.params;
+  const bottom = useSafeAreaInsets().bottom;
+  const selfClient = useSelfClient();
+  const navigation = useNavigation<any>();
+
+  const onSelectDocumentType = (docType: string) => {
+    buttonTap();
+
+    const countryName = getDocumentName(docType);
+
+    selfClient.emit(SdkEvents.DOCUMENT_TYPE_SELECTED, {
+      documentType: docType,
+      documentName: getDocumentNameForEvent(docType),
+      countryCode: countryCode,
+      countryName: countryName,
+    });
+    switch (docType) {
+      case 'p':
+        navigation.navigate('DocumentOnboarding');
+        break;
+      case 'i':
+        navigation.navigate('DocumentOnboarding');
+        break;
+      case 'a':
+        navigation.navigate('AadhaarOnboarding');
+        break;
+      default:
+        navigation.navigate('ComingSoon', { countryCode });
+        break;
+    }
+
+    // TODO: Navigate to the next screen based on document type
+    if (__DEV__) {
+      console.log(
+        `Selected document type: ${docType} for country: ${countryCode}`,
+      );
+    }
+  };
+
+  return (
```

Comment:
nit: oof this screen looks rough imo. not bad for an initial version

<img width="404" height="417" alt="Image" src="https://github.com/user-attachments/assets/e819ead5-babf-48a3-83d1-b9c9b025ebf0" />